src/rasterstats/main.py

```diff
@@ -213,8 +213,26 @@ def gen_zonal_stats(
                     feature_stats['max'] = float(masked.max())
                 if 'mean' in stats:
                     feature_stats['mean'] = float(masked.mean())
+
+                stats_count = int(masked.count())
                 if 'count' in stats:
-                    feature_stats['count'] = int(masked.count())
+                    feature_stats['count'] = stats_count
+
+                if 'area' in stats:
+                    if rast.src is not None:
+                        cellsize = rast.src.res[0] * rast.src.res[1]
+                    else:
+                        cellsize = rast.affine.a * rast.affine.a
```

Comment:
```suggestion
                        cellsize = rast.affine.a * rast.affine.e
```

Reply:
The cells size shouldn't be assumed to be square. It's very rare in practice to see rasters where `a != e` but it could happen

Reply:
I think you need the absolute value here: `abs(rast.affine.a * rast.affine.e)`

Reply:
To be safe, I would recommend reprojecting to be sure. I implemented a utility function for this use-case here - https://github.com/WFP-VAM/prism-app/pull/893/files#diff-0e7a305eb2d7da6652444269a99141f2a18d2fb8ea0ddbc045cd6c5a16e2ee9bR104